### PR TITLE
Make MakeCsvDatasetTest support eager mode

### PR DESCRIPTION
**Pull request by feihugis**
This PR enables `MakeCsvDatasetTest` to run in both `graph` and `eager` modes.

### CODE REVIEW

File: tensorflow/python/data/experimental/kernel_tests/make_csv_dataset_test.py [hunk 1]
```diff
@@ -591,69 +582,65 @@ def str_series(st):
...
+          header=True,
+          shuffle=True,
+          shuffle_seed=5,
+          num_epochs=2,
+      )
+      outputs1 = dataset_ops.make_one_shot_iterator(dataset1).get_next()
+      outputs2 = dataset_ops.make_one_shot_iterator(dataset2).get_next()
+      for _ in range(total_records // batch_size):
+        batch1 = nest.flatten(self.evaluate(outputs1))
+        batch2 = nest.flatten(self.evaluate(outputs2))
```
Comment (rachellim) [inline tensorflow/python/data/experimental/kernel_tests/make_csv_dataset_test.py]:
```suggestion
        batch2 = nest.flatten(self.evaluate(next2()))
```

File: tensorflow/python/data/experimental/kernel_tests/make_csv_dataset_test.py [hunk 2]
```diff
@@ -591,69 +582,65 @@ def str_series(st):
...
+          header=True,
+          shuffle=True,
+          shuffle_seed=6,
+          num_epochs=2,
+      )
+      outputs1 = dataset_ops.make_one_shot_iterator(dataset1).get_next()
+      outputs2 = dataset_ops.make_one_shot_iterator(dataset2).get_next()
+      all_equal = False
+      for _ in range(total_records // batch_size):
+        batch1 = nest.flatten(self.evaluate(outputs1))
```
Comment (rachellim) [inline tensorflow/python/data/experimental/kernel_tests/make_csv_dataset_test.py]: Same suggestions as above.

**Review by rachellim — CHANGES_REQUESTED** [review]
Thanks for the PR! See comments.

**Comment by feihugis** [comment]
@rachellim Thanks for your detailed comments! This PR has been revised accordingly. Could you have another look at the changes?

**Review by rachellim — DISMISSED** [review]

**Review by rachellim — APPROVED** [review]
Thanks!

**Pull request merged**